inc/blocks/ab-test/register.php

```diff
@@ -113,6 +113,34 @@ function render_block( array $attributes, ?string $inner_content = '' ) : string
 		$winner = $results['winner'] ?? false;
 	}
 
+	if ( current_user_can( 'edit_posts' ) ) {
+		if ( is_preview() || isset( $_GET['set_test'] ) ) {
```

Comment:
If we follow the principle of least privilege and we can check a precise post ID we should. This has the benefit of checking post type etc...
```suggestion
	if ( current_user_can( 'edit_post', $post->post_parent ?? 0 ) && ( is_preview() || isset( $_GET['set_test'] ) ) ) {
```

`$post->post_parent` is the ID of the post the XB was created on.